src/WebJobs.Script.WebHost/Diagnostics/SystemLogger.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Azure.WebJobs.Logging;
+using Microsoft.Azure.WebJobs.Script.Config;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.WebJobs.Script.WebHost.Diagnostics
+{
+    public class SystemLogger : ILogger
+    {
+        private readonly bool _isFunctionUserCategory = false;
+        private readonly IEventGenerator _eventGenerator;
+        private readonly ScriptSettingsManager _settingsManager;
+        private readonly string _appName;
+        private readonly string _subscriptionId;
+        private readonly string _categoryName;
+        private readonly string _functionName;
+
+        public SystemLogger(string categoryName, IEventGenerator eventGenerator, ScriptSettingsManager settingsManager)
+        {
+            _settingsManager = settingsManager;
+            _appName = _settingsManager.AzureWebsiteUniqueSlotName;
+            _subscriptionId = Utility.GetSubscriptionId();
+            _eventGenerator = eventGenerator;
+            _categoryName = categoryName;
+            _isFunctionUserCategory = LogCategories.IsFunctionUserCategory(categoryName);
```

Comment:
You know on construction whether this is user logs or not, which makes me wonder why we can't just not create/use the system logger in those cases? I'm probably missing the wide view on how this gets wired up.

Reply:
Yeah, we could only create the logger if it's needed. I'll shift this logic up to the LoggerProvider.